src/profiler.cpp

```diff
@@ -66,92 +63,68 @@ Profiler::Profiler()
 void Profiler::add(const std::string &name, float value)
 {
 	MutexAutoLock lock(m_mutex);
-	{
-		/* No average shall have been used; mark add/max used as -2 */
-		std::map<std::string, int>::iterator n = m_avgcounts.find(name);
-		if (n == m_avgcounts.end()) {
-			m_avgcounts[name] = -2;
-		} else {
-			if (n->second == -1)
-				n->second = -2;
-			assert(n->second == -2);
-		}
-	}
-	{
-		std::map<std::string, float>::iterator n = m_data.find(name);
-		if (n == m_data.end())
-			m_data[name] = value;
-		else
-			n->second += value;
+
+	auto it = m_data.find(name);
+	if (it == m_data.end()) {
+		// No average shall have been used; mark add/max used as -2
+		m_data.emplace(name, DataPair{value, -2});
```

Comment:
Please introduce and use a constant while you're at it. Perhaps `ScopeProfilerType` could be used here (internally only)?